src/main/java/io/supertokens/passwordless/Passwordless.java

```diff
@@ -440,58 +440,84 @@ public static ConsumeCodeResponse consumeCode(TenantIdentifier tenantIdentifier,
+                                    throw new StorageTransactionLogicException(e);
                                 }
-                                throw new StorageQueryException(e);
+                            });
+                            user.loginMethods[0].setVerified(); // newly created user has only one loginMethod
+                        } catch (StorageTransactionLogicException e) {
+                            if (e.actualException instanceof TenantOrAppNotFoundException) {
+                                throw (TenantOrAppNotFoundException) e.actualException;
                             }
+                            throw new StorageQueryException(e);
                         }
-
-                        return new ConsumeCodeResponse(true, user, consumedDevice.email, consumedDevice.phoneNumber, consumedDevice);
-                    } catch (DuplicateEmailException | DuplicatePhoneNumberException e) {
-                        // Getting these would mean that between getting the user and trying creating it:
-                        // 1. the user managed to do a full create+consume flow
-                        // 2. the users email or phoneNumber was updated to the new one (including device cleanup)
-                        // These should be almost impossibly rare, so it's safe to just ask the user to restart.
-                        // Also, both would make the current login fail if done before the transaction
-                        // by cleaning up the device/code this consume would've used.
-                        throw new RestartFlowException();
-                    } catch (DuplicateUserIdException e) {
-                        // We can retry..
                     }
+
+                    return new ConsumeCodeResponse(true, user, consumedDevice.email, consumedDevice.phoneNumber, consumedDevice);
+                } catch (DuplicateEmailException | DuplicatePhoneNumberException e) {
+                    // Getting these would mean that between getting the user and trying creating it:
+                    // 1. the user managed to do a full create+consume flow
+                    // 2. the users email or phoneNumber was updated to the new one (including device cleanup)
+                    // These should be almost impossibly rare, so it's safe to just ask the user to restart.
+                    // Also, both would make the current login fail if done before the transaction
+                    // by cleaning up the device/code this consume would've used.
+                    throw new RestartFlowException();
+                } catch (DuplicateUserIdException e) {
+                    // We can retry..
                 }
             }
         } else {
-            // We do not need this cleanup if we are creating the user, since it uses the email/phoneNumber of the
-            // device, which has already been cleaned up
             if (setEmailVerified && consumedDevice.email != null) {
```

Comment:
the suff in the else branch needs to go away since it will be done in verifyCode

Reply:
I don't think this is applicable anymore.